AsyncDisplayKit/Details/ASChangeSetDataController.mm

```diff
@@ -81,8 +81,16 @@ - (void)endUpdatesAnimated:(BOOL)animated completion:(void (^)(BOOL))completion
     
     ASDataControllerLogEvent(self, @"triggeredUpdate: %@", _changeSet);
     
+    // TODO: move
+    [super prepareMoveItemChanges:[_changeSet itemChangesOfType:_ASHierarchyChangeTypeMove]];
+    
+    // Prepare UIKit batch updates
     [super beginUpdates];
     
+    for (_ASHierarchyMoveItemChange *change in [_changeSet itemChangesOfType:_ASHierarchyChangeTypeMove]) {
+      [super moveRowAtIndexPath:change.fromIndexPath toIndexPath:change.toIndexPath withAnimationOptions:change.animationOptions];
+    }
+    
```

Comment:
Hmm this still won't work, I believe. For the reasons in [this comment](https://github.com/facebook/AsyncDisplayKit/pull/2868#discussion_r94865245), `ASDataController` simply _cannot implement_ `moveRowAtIndexPath:` because the move has to be decomposed into delete and insert, which have to be performed separately, each with all the other operations of the same kind.

I believe what we should do instead is modify ASDataController to have a method like `insertRowsAtIndexPaths:withExistingNodes:…` that takes a `NSDictionary<NSIndexPath, ASIndexedNodeContext>`. Before the update, we build that dictionary by going through all the `_ASHierarchyMoveItemChange` and using the `from` index path to fetch the node context and the `to` index path to act as the key into the dictionary.

Then inside `insertRows`, the data controller will always check the provided dictionary to see if there's already a context for the given index path, and use that if possible.

* Note we could use ASCellNodes directly, or contexts, whichever is more convenient.

Does that make sense?